src/algebra/big_operators/fin.lean

```diff
@@ -206,7 +206,33 @@ begin
 end
 
 end partial_prod
+section mul_nth
 
+variables [has_mul α] {n : ℕ}
+
+/-- Sends `(g₀, ..., gₙ)` to `(g₀, ..., gⱼgⱼ₊₁, ..., gₙ)`. -/
+@[to_additive "Sends `(g₀, ..., gₙ)` to `(g₀, ..., gⱼ + gⱼ₊₁, ..., gₙ)`."]
+def mul_nth (j : ℕ) (g : fin (n + 1) → α) (k : fin n) : α :=
+if (k : ℕ) < j then g (cast_lt k (lt_trans k.2 $ lt_add_one _)) else
+if (k : ℕ) = j then g (cast_lt k (lt_trans k.2 $ lt_add_one _)) * g (add_nat 1 k)
+else g (add_nat 1 k)
```

Comment:
I would suggest using the standard functions on `Fin` types for this definition (and the following equational lemmas):
```suggestion
@[to_additive "Sends `(g₀, ..., gₙ)` to `(g₀, ..., gⱼ + gⱼ₊₁, ..., gₙ)`."]
def mul_nth (j : ℕ) (g : fin (n + 1) → α) (k : fin n) : α :=
if (k : ℕ) < j then g k.cast_succ else
if (k : ℕ) = j then g k.cast_succ * g k.succ
else g k.succ
```
(I will try to look at the rest of the PR tomorrow.)

Reply:
Otherwise, it looks very good to me!

Reply:
thank you!